src/server/actions/s3.actions.ts

```diff
@@ -0,0 +1,35 @@
+'use server'
+
+import { getS3BrowserClient, getS3Client, s3BucketName } from '@/server/aws'
+import { MinimalJobInfo, minimalJobInfoSchema } from '@/lib/types'
+import { pathForStudyJobCode } from '@/lib/paths'
+import { createPresignedPost } from '@aws-sdk/s3-presigned-post'
+import { userAction, z } from '@/server/actions/wrappers'
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner'
+import { DeleteObjectCommand } from '@aws-sdk/client-s3'
+
+export const signedUrlForCodeUploadAction = userAction(async (jobInfo: MinimalJobInfo) => {
+    const prefix = pathForStudyJobCode(jobInfo)
+
+    return await createPresignedPost(getS3BrowserClient(), {
+        Bucket: s3BucketName(),
+        Conditions: [['starts-with', '$key', prefix]],
+        Expires: 3600,
+        Key: prefix + '/${filename}', // single quotes are intentional, S3 will replace ${filename} with the filename
+    })
+}, minimalJobInfoSchema)
+
+export const signedUrlForStudyFileUploadAction = userAction(async (path: string) => {
+    return await createPresignedPost(getS3BrowserClient(), {
+        Bucket: s3BucketName(),
+        Expires: 3600,
+        Conditions: [['starts-with', '$key', path]],
+        Key: path + '/${filename}', // single quotes are intentional, S3 will replace ${filename} with the filename
+    })
+}, z.string())
+
+export const signedUrlForDeletingStudyFilesAction = userAction(async (Key: string) => {
+    return await getSignedUrl(getS3Client(), new DeleteObjectCommand({ Bucket: s3BucketName(), Key }), {
+        expiresIn: 3600,
+    })
+}, z.string())
```

Comment:
I don't think these should be actions,  they can just be functions that the the study save will call to generate all these and send to the client after the study's saved.  

I also see why we need a different DELETE url now  so disregard that comment above

Reply:
I don't think these are used any longer right?

Reply:
ah yes! great catch i forgot to delete this file